### PR TITLE
API validation test

### PR DESCRIPTION
**Pull request by igoihman**
**What this PR does / why we need it**:
e2e testing of api object validation

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by igoihman** [comment]
CI failures are related to basic_sanity_test.
API Validation tests pass on CI.

**Comment by rollandf** [comment]
:+1:

File: tests/api_validation_test.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+					dv["spec"].(map[string]interface{})["source"] = map[string]interface{}{"s3": map[string]interface{}{"url": url}}
+				case "pvc":
+					namespace := args[0]
+					name := args[1]
+					dv["spec"].(map[string]interface{})["source"] = map[string]interface{}{
+						"pvc": map[string]interface{}{
+							"namespace": namespace,
+							"name":      name}}
+				}
+
```
Comment (zvikorn) [inline tests/api_validation_test.go]: Don't you want to check all other sources (none, registry, blank, upload)?

File: tests/api_validation_test.go [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+	"github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+
+	"kubevirt.io/containerized-data-importer/tests/framework"
+	"kubevirt.io/containerized-data-importer/tests/utils"
+)
+
+const (
+	dataVolumeName     = "test-dv"
+	validURL           = "http://www.example.com/example.img"
```
Comment (zvikorn) [inline tests/api_validation_test.go]: If you are going to change the code per my comment below about other source types, then please change validURL to validURLFormat too. If you are not, then that's ok. It's not critical of course

File: tests/api_validation_test.go (hunk 1 above)
Comment (igoihman) [reply tests/api_validation_test.go]: I added the tests according to the test plan.

Comment (zvikorn) [reply tests/api_validation_test.go]: We can merge this and then update the test case and this code.

**Pull request merged**